03-rnaseq/dimension_reduction_rnaseq_01_pca.Rmd

```diff
@@ -0,0 +1,368 @@
+  design = ~1 # Here we are not specifying a model -- Replace with an appropriate design variable for your analysis
+)
+```
+
+## Define a minimum counts cutoff
+
+We want to filter out the genes that have not been expressed or that have low expression counts because we want to remove any possible noise from our data before we normalize the data and create our plot.
+We are going to do some pre-filtering to keep only genes with 10 or more reads total.
+Note that rows represent gene data and the columns represent sample data in our dataset.
+
+```{r}
+# Let's take a look at the distribution of counts first
+
+# Sum up the counts for each gene -- which means we want the sum of counts for each row as the rows represent gene data and the columns represent sample data in our dataset
+count_sums <- data.frame(count = rowSums(counts(dds)))
+
+# Create a density plot using the sums
+ggplot2::ggplot(count_sums, ggplot2::aes(x = count)) +
+  ggplot2::geom_density() +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = 10), # This line represents the current cutoff of 10
+    color = "blue",
+    linetype = "solid"
+  ) +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = mean(count)), # This line represents the mean count value
+    color = "red",
+    linetype = "dashed"
+  )
+```
+
+It appears that our arbitrary cutoff of 10 (the solid blue line) is a decent cutoff to make here as it seems to be just about inline with the mean counts value (the red dashed line) and retains the data points that are not in the majority and therefore may hold significance.
+
+```{r}
+# Define a minimum counts cutoff and filter `DESeqDataSet` object to include
+# only rows that have counts above the cutoff
+genes_to_keep <- rowSums(counts(dds)) >= 10
+dds <- dds[genes_to_keep, ]
+```
+
+## Perform DESeq2 normalization
+
+We are going to use the `vst()` function from the `DESeq2` package to normalize the data.
+TODO: Add a note/link to section in documentation on why we use `vst()` vs `rlog()` and `normTransform()`.
+
+```{r}
+# Normalize the data in the `DESeqDataSet` object using the `vst()` function from the `DESEq2` R package
+dds_norm <- vst(dds)
+```
+
+
+## Create PCA plot using DESeq2
+
+In this code chunk, the variable `refinebio_treatment` is given to the `plotPCA()` function as part of the goal of the experiment is to analyze the differential responses to androgen deprivation therapy (ADT).
+
+```{r}
+plotPCA(dds_norm,
+  intgroup = "refinebio_treatment"
+)
+```
+
+In this chunk, we are going to add a variable to our plot for annotation.
+
+The variables `refinebio_treatment` and `refinebio_disease` are given to the `plotPCA()`
+due to the hypothesis of the experiment stating "that differences in tumor transcriptional programs may reflect differential responses to androgen deprivation therapy (ADT) and subsequent metastasis".
+
+```{r}
+plotPCA(dds_norm,
+  intgroup = c("refinebio_treatment", "refinebio_disease")
```

Comment:
I would put a comment to highlight the difference here.

Reply:
_Believe_ I added such comment here in the second to last commit.